client/src/app/core/ui-services/base-sort-list.service.ts

```diff
@@ -98,20 +103,26 @@ export abstract class BaseSortListService<V extends BaseViewModel> {
      * @returns the current sorting property
      */
     public get sortProperty(): keyof V {
-        return this.sortDefinition.sortProperty;
+        if (this.sortDefinition) {
+            return this.sortDefinition.sortProperty;
+        } else {
+            return null;
```

Comment:
See above: Correct type annotations and add null checks everywhere this function is called.